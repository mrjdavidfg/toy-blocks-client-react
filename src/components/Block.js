import React from "react";
import PropTypes from "prop-types";
import { makeStyles, Typography } from "@material-ui/core";

const Block = ({ block }) => {
  const classes = useStyles();

  const { attributes } = block;

  return (
    <div className={classes.root}>
      <div className={classes.index}>{attributes.index}</div>
      <Typography variant="body2" className={classes.text}>
        {attributes.data}
      </Typography>
    </div>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    padding: "8px 8px 4px",
    background: "rgba(0, 0, 0, 0.12)",
    borderRadius: 2,
  },
  index: {
    fontSize: theme.typography.pxToRem(10),
    lineHeight: "16px",
    fontWeight: 700,
    color: "#304FFE",
  },
  text: {
    lineHeight: "20px",
  },
}));

Block.propTypes = {
  block: PropTypes.object,
};

export default Block;
