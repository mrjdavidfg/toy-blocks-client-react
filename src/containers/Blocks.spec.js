import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import ConnectedBlocks, { Blocks } from "./Blocks";
import Block from "../components/Block";

describe("<Blocks />", () => {
  const actions = {
    loadBlocks: jest.fn(),
  };

  const node = {
    url: "https://thawing-springs-53971.herokuapp.com",
    online: false,
    name: "Node 1",
    loading: false,
  };

  const blocks = {
    "https://thawing-springs-53971.herokuapp.com": {
      loading: false,
      error: false,
      list: [
        {
          id: 123,
          type: "blocks",
          attributes: {
            index: 1,
            timestamp: 1530674152,
            data: "Block 1",
            "previous-hash": "KsmmdGrKVDr43/OYlM/oFzr7oh6wHG+uM9UpRyIoVe8=",
            hash: "udfJLmh13UNAxG4F/1on07OMN1K1vCuaTYn9H2XGiX0=",
          },
        },
      ],
    },
  };

  it("should contain <Block />", () => {
    const wrapper = shallow(
      <Blocks actions={actions} node={node} blocks={blocks} />
    );

    expect(wrapper.find(Block).length).toEqual(1);
  });

  it("should match snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)({
      nodes: { list: [node] },
      blocks,
    });
    const component = create(
      <Provider store={store}>
        <ConnectedBlocks node={node} />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
