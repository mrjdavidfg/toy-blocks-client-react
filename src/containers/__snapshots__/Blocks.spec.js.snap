// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blocks /> should match snapshot 1`] = `
<div
  style={
    Object {
      "display": "grid",
      "gridGap": 8,
      "width": "100%",
    }
  }
>
  <div
    className="makeStyles-root-1"
  >
    <div
      className="makeStyles-index-2"
    >
      1
    </div>
    <p
      className="MuiTypography-root makeStyles-text-3 MuiTypography-body2"
    >
      Block 1
    </p>
  </div>
</div>
`;
